src/main/java/io/syndesis/verifier/v1/ActionPropertiesEndpoint.java

```diff
@@ -0,0 +1,82 @@
+/**
+ * Copyright (C) 2016 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.syndesis.verifier.v1;
+
+import java.util.Map;
+
+import javax.ws.rs.Consumes;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+
+import io.syndesis.verifier.v1.metadata.MetadataAdapter;
+
+import org.apache.camel.CamelContext;
+import org.apache.camel.component.extension.MetaDataExtension;
+import org.apache.camel.component.extension.MetaDataExtension.MetaData;
+import org.apache.camel.impl.DefaultCamelContext;
+import org.springframework.stereotype.Component;
+
+@Component
+@Path("/action/properties")
+public class ActionPropertiesEndpoint {
+
+    private final Map<String, MetadataAdapter> adapters;
+
+    public ActionPropertiesEndpoint(final Map<String, MetadataAdapter> adapters) {
+        this.adapters = adapters;
+    }
+
+    @POST
+    @Produces(MediaType.APPLICATION_JSON)
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Path("/{id}")
+    public Object properties(@PathParam("id") final String connectorId, final Map<String, Object> properties)
+        throws Exception {
+        final MetadataAdapter adapter = adapters.get(connectorId + "-adapter");
+
+        if (adapter == null) {
+            throw null;
+        }
+
+        final CamelContext camel = camelContext();
+        camel.start();
+
+        try {
+            final MetaData metaData = fetchMetadata(camel, connectorId, properties);
+
+            return adapter.apply(properties, metaData);
+        } finally {
+            camel.stop();
+        }
+    }
+
+    protected DefaultCamelContext camelContext() {
+        return new DefaultCamelContext();
+    }
+
+    static MetaData fetchMetadata(final CamelContext camel, final String connectorId,
+        final Map<String, Object> properties) {
+        final MetaDataExtension metadataExtension = camel.getComponent(connectorId, true, false)
+            .getExtension(MetaDataExtension.class).orElseThrow(
+                () -> new IllegalArgumentException("No Metadata extension present for connector: " + connectorId));
+
+        return metadataExtension.meta(properties)
```

Comment:
my question in the other comment probably boils down how this method decide on which properties which `scope` to return. but it all can be likely that I don't understand the real flow yet.

I mean, at some point the meta data extension of a connector has to decide what to return in which scope. This obviously depends on the input parameters coming in from syndesis-rest. And we probably need to 'standardize' this, at least in the adapters, to react appropriately on the same kind of input params.